src/images/avatar.ts

```diff
@@ -8,6 +9,36 @@ import {Component, Input, ChangeDetectionStrategy} from 'angular2/core';
 export class NglAvatar {
   @Input() src: string = '';
   @Input() alt: string = '';
-  @Input() size: 'x-small' | 'small' | 'medium' | 'large' = 'medium';
-  @Input() type: 'rectangle' | 'circle' = 'rectangle';
+  @Input() size: 'x-small' | 'small' | 'medium' | 'large';
+
+  @Input('size') set setSize(value: 'x-small' | 'small' | 'medium' | 'large') {
+    this.updateClass(this._size, value);
+    this._size = value;
+  }
+
+  @Input('type') set setType(value: 'rectangle' | 'circle') {
+    this.updateClass(this._type, value);
+    this._type = value;
+  }
+
+  private _type: 'rectangle' | 'circle';
+  private _size: 'x-small' | 'small' | 'medium' | 'large';
+
+  constructor(public element: ElementRef, public renderer: Renderer) {
+    renderer.setElementClass(element.nativeElement, 'slds-avatar', true);
+  }
+
+  ngOnInit() {
+    if (!this._type) {
+      this.renderer.setElementClass(this.element.nativeElement, 'slds-avatar--rectangle', true);
+    }
+
+    if (!this._size) {
+      this.renderer.setElementClass(this.element.nativeElement, 'slds-avatar--medium', true);
+    }
+  }
+
+  private updateClass(oldValue: string, newValue: string) {
+    replaceClass(this, `slds-avatar--${oldValue}`, newValue ? `slds-avatar--${newValue}` : '');
+  }
 };
```

Comment:
OK, I have a final idea. If this doesn't work I'm out of ideas on this one! 😃 Could we do ```@Input('class')``` so as to get the updated external classes? Again I cannot test this right now and I haven't tried this before (it seems like a loop binding to an input so we could use a setter+flag or something similar) but if we could do it while keeping it simple it would be nice! Otherwise, I'd suggest sticking to what we have and moving on since it works nicely as it is. 👍